### PR TITLE
Complexity Updates

### PR DESCRIPTION
**Pull request by wfleming**
* Qualify `max-statements` (long methods) as complexity
* Update remediation points calculation to reflect RuboCop base point/per-violation values
* Pull threshold values from actual config for the purposes of calculating remediation points.

:eyeglasses: @codeclimate/review

### CODE REVIEW

**Comment by wfleming** [comment]
Note the changed eslint config turns up a complexity complaint in the `bin/eslint` code: i'm inclined to leave that as is for now.

File: lib/checks.js [hunk 1]
```diff
@@ -113,21 +114,48 @@ var categories = function(checkName) {
...
 
-var remediationPoints = function(checkName, message) {
-  if (checkName === "complexity") {
+// Here Be Dragons: this function extracts the relevant value that triggered the issue for
+// checks in the Complexity category. Unfortunately, these values are not available in a
+// structured way, so we extract them from strings. That means that any check categorized
+// as Complexity MUST have a rule here to extract value.
+//
+// If a matching string segment cannot be found, `null` will be returned.
+var messageMetricValue = function(message) {
```
Comment (GordonDiggs) [inline lib/checks.js]: Could you add some tests for this function?

File: lib/checks.js [hunk 2]
```diff
@@ -113,21 +114,48 @@ var categories = function(checkName) {
...
+    return parseInt(match[1], 10);
+  }
+  return null;
+};
+
+var metricThreshold = function(ruleId, eslintConfig) {
+   return eslintConfig.rules[ruleId][1];
+};
+
+var remediationPoints = function(checkName, message, eslintConfig) {
```
Comment (GordonDiggs) [inline lib/checks.js]: Could you add some tests here too?

Comment (wfleming) [inline lib/checks.js]: On the one hand: this means finally adding some tests for this engine, which makes me happy. On the other hand, this means I'm responsible for setting up the testing setup on this engine, which makes me sad.

But I agree it's probably worth testing stuff here. I'll take a crack at it.

**Comment by wfleming** [comment]
@GordonDiggs tests have been added if you'd like to take a look!

@ABaldwinHunter this seems relevant to your recent interests (viz. your lunch and learn talk), so it'd be helpful if you took a look as well.

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,9 @@
-.PHONY: image
+.PHONY: image test
 
 IMAGE_NAME ?= codeclimate/codeclimate-eslint
 
 image:
 	docker build --rm -t $(IMAGE_NAME) .
+
+test: image
+	docker run $(IMAGE_NAME) node_modules/.bin/mocha
```
Comment (GordonDiggs) [inline Makefile]: To be a bit more conventional, can you specify a task inside the `package.json` file? Something like:

```json
"scripts": {
    "test": "mocha test"
  }
```

And then this invocation can be `docker run $(IMAGE_NAME) npm run test`

**Comment by GordonDiggs** [comment]
One comment but otherwise this looks great! Excited to actually have tests here now.

File: Makefile (hunk 3 above)
Comment (wfleming) [inline Makefile]: Good suggestion: done.

**Comment by GordonDiggs** [comment]
:+1:

**Comment by wfleming** [comment]
As noted above, the CC issue is a function complexity thing in `bin/eslint` that's been there a while: I think that's out of scope to fix now, so I'm going to mercy merge.

**Pull request merged**